lib/src/piece/variable.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import '../back_end/code_writer.dart';
+import '../constants.dart';
+import 'piece.dart';
+
+/// A variable declaration.
+///
+/// Used for local variable declaration statements, top-level variable
+/// declarations and field declarations.
+///
+/// Typed and untyped variables have slightly different splitting logic.
+/// Untyped variables never split after the keyword but do indent subsequent
+/// variables:
+///
+/// ```
+/// var longVariableName = initializer,
+///     anotherVariable = anotherInitializer;
+/// ```
+///
+/// Typed variables can split that way too:
+///
+/// ```
+/// String longVariableName = initializer,
+///     anotherVariable = anotherInitializer;
+/// ```
+///
+/// But they can also split after the type annotation. When that happens, the
+/// variables aren't indented:
+///
+/// ```
+/// VeryLongTypeName
+/// longVariableName = initializer,
+/// anotherVariable = anotherInitializer;
+/// ```
+class VariablePiece extends Piece {
+  /// Split between each variable in a multiple variable declaration.
+  static const State _betweenVariables = State(1);
+
+  /// Split after the type annotation and between each variable.
+  static const State _afterType = State(2);
```

Comment:
[optional]

```suggestion
  static const _betweenVariables = State(1);

  /// Split after the type annotation and between each variable.
  static const _afterType = State(2);
```

Reply:
I'm not sure if it's habit or an actually good idea, but my consistent practice is to always type annotate members. <shrug>

Reply:
> consistent practice

https://github.com/dart-lang/dart_style/pull/1279/files#diff-dc088a2965c2b639564656129400eaad08b4e1c424d03bd272d941431146aaceR85-R91

😝